---
title: "Tutorial: SemiBin"
permalink: tutorials/2022-06-14-semibin/
meta: SemiBin tutorial
---

**tl;dr**: Learn how to use
[SemiBin](https://github.com/BigDataBiology/SemiBin/) from the original
developers. In particular, we will see how to run SemiBin on a set of of human
gut samples with different binning modes.

### Prerequisites

What you should _know_/_be able to do_ **before** the tutorial:

- You should know what a contig and BAM file is.
- You should be able to run programmes on the command line
- You should be able to install the SemiBin. You can see the details from [How to install](https://semibin.readthedocs.io/en/latest/install/).

You will also be sent some information a few days before the tutorial with both the signup links &amp; and some requests to prepare for the event. You should expect to spend some time (10-15 minutes) installing the software and downloading the tutorial data.

[SIGN UP LINK](https://bit.ly/2022-06-14_semibin-tutorial)

### Learning objectives

- What is the metagenomic binning and what is the basic idea of existing binning tools?
- The basic idea behind SemiBin.
- How to get MAGs from SemiBin with different binning modes.
- We will not cover functional analysis of MAGs.

### Location &amp; Schedule

**Date**: 14th June, 2022 @ Noon UTC ([check your timezone](https://everytimezone.com/s/c8221019))

**Location**: Zoom &amp; Discord (links will be sent to registered participants)

## Background information

- SemiBin webpage: [https://github.com/BigDataBiology/SemiBin/](https://github.com/BigDataBiology/SemiBin/)
- SemiBin manuscript: _Pan et al._ [A deep siamese neural network improves metagenome-assembled genomes in microbiome datasets across different environments](https://www.nature.com/articles/s41467-022-29843-y), in **Nature Communications** (2022)

